workflow-bot-app/src/test/java/com/symphony/bdk/workflow/ExecuteRequestIntegrationTest.java

```diff
@@ -0,0 +1,76 @@
+package com.symphony.bdk.workflow;
+
+import static com.symphony.bdk.workflow.custom.assertion.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.symphony.bdk.http.api.ApiClient;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiResponse;
+import com.symphony.bdk.http.api.util.TypeReference;
+import com.symphony.bdk.workflow.swadl.SwadlParser;
+import com.symphony.bdk.workflow.swadl.v1.Workflow;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+import java.util.Map;
+
+class ExecuteRequestIntegrationTest extends IntegrationTest {
+
+  private static final String OUTPUTS_STATUS_KEY = "%s.outputs.status";
+  private static final String OUTPUTS_BODY_KEY = "%s.outputs.body";
+
+  @Test
+  void executeRequestSuccessful() throws Exception {
+    final Workflow workflow =
+        SwadlParser.fromYaml(getClass().getResourceAsStream("/request/execute-request.swadl.yaml"));
+
+    final ApiClient mockedApiClient = mock(ApiClient.class);
+    final Map<String, String> header = Map.of("headerKey", "headerValue");
+    final Map<String, Object> body = Map.of("args", Map.of("key", "value"));
+    final String jsonResponse =
+        "{\"name\":\"john\",\"age\":22,\"contact\":{\"phone\":\"0123456\",\"email\":\"john@symphony.com\"}}";
+
+    final ApiResponse<Object> mockedResponse = new ApiResponse<>(200, Collections.emptyMap(), jsonResponse);
+
+    when(bdkGateway.apiClient(anyString())).thenReturn(mockedApiClient);
+    when(mockedApiClient.invokeAPI(eq(""), eq("GET"), anyList(), eq(body), eq(header), anyMap(), anyMap(),
+        eq("application/json"), eq("application/json"), eq(null), any(TypeReference.class)))
+        .thenReturn(mockedResponse);
+
+    engine.deploy(workflow);
+
+    engine.onEvent(messageReceived("/execute"));
+
+    assertThat(workflow).isExecuted()
+        .hasOutput(String.format(OUTPUTS_STATUS_KEY, "executeGetRequest"), 200)
+        .hasOutput(String.format(OUTPUTS_BODY_KEY, "executeGetRequest"), jsonResponse);
```

Comment:
Could you also unit test that you can retrieve a specific field of the response? Following the swadl example: 
```yaml
id: exchange-rates-workflow
activities:
  - execute-request:
      id: callExchangeRatesApi
      method: GET
      url: https://open.er-api.com/v6/latest/EUR
      on:
        message-received:
          content: /show-exchange-rates

  - send-message:
      id: sendRates
      content: EURO= ${callExchangeRatesApi.outputs.body.rates['EUR']}, USD= ${callExchangeRatesApi.outputs.body.rates['USD']}
```

here we would ensure that it's possible to retrieve the `name` value

Reply:
There was nothing to do when the request succeeds. When an ApiException is catched, I map the responseBody (which is a string) to a map before storing it in outputs. Thus, we can process it in SWADL.
For tests, I added a script activity that checks the output values (...body.contact.phone and others). It makes the test fail if the actual value is not expected or not found.